chia/data_layer/data_layer_server.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict
+
+from aiohttp import web
+
+from chia.data_layer.download_data import is_filename_valid
+from chia.server.upnp import UPnP
+from chia.util.path import path_from_root
+
+
+@dataclass
+class DataLayerServer:
+    root_path: Path
+    config: Dict[str, Any]
+    log: logging.Logger
+
+    async def start(self) -> None:
+        self.log.info("Starting Data Layer Server.")
+        self.port = self.config["host_port"]
+
+        # Setup UPnP for the data_layer_service port
+        self.upnp: UPnP = UPnP()
+        self.upnp.remap(self.port)
+
+        server_files_replaced: str = self.config.get(
+            "server_files_location", "data_layer/db/server_files_location_CHALLENGE"
+        ).replace("CHALLENGE", self.config["selected_network"])
+        self.server_dir = path_from_root(self.root_path, server_files_replaced)
+
+        app = web.Application()
+        app.add_routes([web.get("/{filename}", self.file_handler)])
+        self.runner = web.AppRunner(app)
+        await self.runner.setup()
+        self.site = web.TCPSite(self.runner, self.config["host_ip"], port=self.port)
+        await self.site.start()
+        self.log.info("Started Data Layer Server.")
+
+    async def stop(self) -> None:
+        self.upnp.release(self.port)
+        # this is a blocking call, waiting for the UPnP thread to exit
```

Comment:
```suggestion
        # upnp.release() is a blocking call, waiting for the UPnP thread to exit
```

Reply:
Pretty sure this does refer to the shutdown call below it.  Clarified in https://github.com/Chia-Network/chia-blockchain/pull/13216.